src/main/java/org/jabref/logic/importer/fetcher/MathSciNet.java

```diff
@@ -102,34 +117,137 @@ public URL getUrlForIdentifier(String identifier) throws URISyntaxException, Mal
+            } catch (JSONException | ParseException e) {
                 LOGGER.error("An error occurred while parsing fetched data", e);
                 throw new ParseException("Error when parsing entry", e);
             }
             return entries;
         };
     }
 
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            BibEntry entry = new BibEntry(StandardEntryType.Article);
+
+            // Set the author and keywords field
+            Optional<String> authors = toAuthors(item.optJSONArray("authors"));
+            authors.ifPresent(value -> entry.setField(StandardField.AUTHOR, value));
+
+            Optional<String> keywords = Optional.ofNullable(getKeywords(item.optJSONObject("primaryClass")));
+            keywords.ifPresent(value -> entry.setField(StandardField.KEYWORDS, value));
+
+            // Set the rest of the fields based on the mappings
+            for (Map.Entry<StandardField, List<String>> mapEntry : FIELD_MAPPINGS.entrySet()) {
+                StandardField field = mapEntry.getKey();
+                List<String> path = mapEntry.getValue();
+
+                // Skip author and keywords fields as they are already set
+                if (field == StandardField.AUTHOR || field == StandardField.KEYWORDS) {
+                    continue;
+                }
+
+                Optional<String> value = getOthers(item, path);
+                value.ifPresent(v -> entry.setField(field, v));
+            }
+
+            // Handle articleUrl and mrnumber fields separately, as they are non-nested properties in the JSON and can be retrieved as Strings directly
+            String doi = item.optString("articleUrl", "");
+            if (!doi.isEmpty()) {
+                entry.setField(StandardField.DOI, doi);
+            }
+
+            String mrNumber = item.optString("mrnumber", "");
+            if (!mrNumber.isEmpty()) {
+                entry.setField(StandardField.MR_NUMBER, mrNumber);
+            }
+
+            return entry;
+        } catch (JSONException exception) {
+            throw new ParseException("MathSciNet API JSON format has changed", exception);
+        }
+    }
+
+    private Optional<String> toAuthors(JSONArray authors) {
+        if (authors == null) {
+            return Optional.empty();
+        }
+
+        String authorsString = IntStream.range(0, authors.length())
+                                        .mapToObj(authors::getJSONObject)
+                                        .map(author -> {
+                                            String name = author.optString("name", "");
+                                            return fixStringEncoding(name);
+                                        })
+                                        .collect(Collectors.joining(" and "));
+
+        return Optional.of(authorsString);
+    }
+
+    private String getKeywords(JSONObject primaryClass) {
```

Comment:
Please also use the `Optional<String>` thing you did at `toAuhtors`. Because of consistency.